### PR TITLE
Improving properties metadata

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
This PR introduces a new PropertyMetadata and PropertySet object for recording what thermophysical properties are supported by a property package. The eventual goal is to have a concrete definition of the standard properties used in IDAES that can be accessed programmatically and does not rely on string constants, as well as formalizing what metadata is stored for each.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by andrewlee94** [comment]
Note that test failures are expected for now - there are some tests that are using the property metadata for unexpected purposes.

File: idaes/core/base/property_set.py [hunk 1]
```diff
@@ -0,0 +1,2099 @@
...
+    def __iter__(self):
+        for a in dir(self):
+            if not a.startswith("_"):
+                # Ignore anything starting with "_" as they will show up twice otherwise
+                # This will also skip private attributes of object class
+                aobj = getattr(self, a)
+                if isinstance(aobj, PropertyMetadata):
+                    yield aobj
+
+    def define_property(
```
Comment (dangunter) [inline idaes/core/base/property_set.py:200]: The docs say name is required, yet the code seems to silently use the default value of None. Suggest taking `=None` off the name argument to enforce this.

**Review by eslickj — APPROVED** [review]

**Comment by ksbeattie** [comment]
@andrewlee94 is this needed for the Nov release?

**Comment by andrewlee94** [comment]
@ksbeattie No, it is not required (although it would be nice to get it finished).

**Comment by eslickj** [comment]
I found another problem.  Producing hydrogen lowers the cooling duty in the NGCC + SOEC since it lower the condenser duty by a lot since steam is extracted from the steam cycle.  I need to fix some cooling duty parameters at the NGCC (no SEOC) values to get a cooling system sized for the SOEC to be able to turn off.  Working on that now.

**Review by jsiirola — APPROVED** [review]

**Pull request merged**